### DISCUSSION TITLE
divide by 0 error?

### DISCUSSION THREAD

**Discussion by xEchoes** [FabricMC/fabric-api#1914] [Player support]
every single time i join a server with fabric, it instantly crashes with a divide by zero error, any help?

---- Minecraft Crash Report ----
// There are four lights!

Time: 12/19/21, 2:28 PM
Description: Unexpected error

java.lang.ArithmeticException: / by zero
	at net.minecraft.class_761.method_22710(class_761.java:1153)
	at net.minecraft.class_757.method_3188(class_757.java:1026)
	at net.minecraft.class_757.method_3192(class_757.java:810)
	at net.minecraft.class_310.method_1523(class_310.java:1112)
	at net.minecraft.class_310.method_1514(class_310.java:728)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Render thread
Stacktrace:
	at net.minecraft.class_761.method_22710(class_761.java:1153)
	at net.minecraft.class_757.method_3188(class_757.java:1026)

-- Affected level --
Details:
	All players: 4 total; [class_746['xEchoes'/530831, l='ClientLevel', x=-3.51, y=106.00, z=-38.48], class_745['§6§lGAIA'/243, l='ClientLevel', x=5.50, y=92.00, z=7.47], class_745['§9§lWEBSITE'/240, l='ClientLevel', x=-0.53, y=92.00, z=9.47], class_745['§a§lEARTH'/246, l='ClientLevel', x=-6.50, y=92.00, z=7.44]]
	Chunk stats: 65536, 390
	Level dimension: minecraft:overworld
	Level spawn location: World: (-4,106,-39), Section: (at 12,10,9 in -1,6,-3; chunk contains blocks -16,0,-48 to -1,255,-33), Region: (-1,-1; contains chunks -32,-32 to -1,-1, blocks -512,0,-512 to -1,255,-1)
	Level time: 37340955 game time, 37358346 day time
	Server brand: BungeeCord (git:BungeeCord-Bootstrap:1.18-R0.1-SNAPSHOT:a9d75c5:1616) <- PaperSpigot
	Server type: Non-integrated multiplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:370)
	at net.minecraft.class_310.method_1587(class_310.java:2399)
	at net.minecraft.class_310.method_1514(class_310.java:752)
	at net.minecraft.client.main.Main.main(Main.java:217)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:78)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:567)
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608)
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77)
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23)

-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: Default, Fabric Mods

-- System Details --
Details:
	Minecraft Version: 1.17.1
	Minecraft Version ID: 1.17.1
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 16.0.1, Microsoft
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Microsoft
	Memory: 865238144 bytes (825 MiB) / 1644167168 bytes (1568 MiB) up to 2147483648 bytes (2048 MiB)
	CPUs: 12
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Ryzen 5 3600 6-Core Processor              
	Identifier: AuthenticAMD Family 23 Model 113 Stepping 0
	Microarchitecture: Zen 2
	Frequency (GHz): 3.59
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: Radeon RX 580 Series
	Graphics card #0 vendor: Advanced Micro Devices, Inc. (0x1002)
	Graphics card #0 VRAM (MB): 4095.00
	Graphics card #0 deviceId: 0x67df
	Graphics card #0 versionInfo: DriverVersion=30.0.13023.1012
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.13
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 8192.00
	Memory slot #1 clockSpeed (GHz): 2.13
	Memory slot #1 type: DDR4
	Memory slot #2 capacity (MB): 8192.00
	Memory slot #2 clockSpeed (GHz): 2.13
	Memory slot #2 type: DDR4
	Memory slot #3 capacity (MB): 8192.00
	Memory slot #3 clockSpeed (GHz): 2.13
	Memory slot #3 type: DDR4
	Virtual memory max (MB): 45002.89
	Virtual memory used (MB): 35175.32
	Swap memory total (MB): 12288.00
	Swap memory used (MB): 778.55
	JVM Flags: 9 total; -XX:HeapDumpPath=MojangTricksIntelDriversForPerformance_javaw.exe_minecraft.exe.heapdump -Xss1M -Xmx2G -XX:+UnlockExperimentalVMOptions -XX:+UseG1GC -XX:G1NewSizePercent=20 -XX:G1ReservePercent=20 -XX:MaxGCPauseMillis=50 -XX:G1HeapRegionSize=32M
	Fabric Mods: 
		fabric: Fabric API 0.44.0+1.17
		fabric-api-base: Fabric API Base 0.4.0+cf39a74318
		fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.4.0+cb9053d118
		fabric-biome-api-v1: Fabric Biome API (v1) 3.2.2+cf39a74318
		fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.6+cf39a74318
		fabric-command-api-v1: Fabric Command API (v1) 1.1.4+cf39a74318
		fabric-commands-v0: Fabric Commands (v0) 0.2.3+cf39a74318
		fabric-containers-v0: Fabric Containers (v0) 0.1.13+cf39a74318
		fabric-content-registries-v0: Fabric Content Registries (v0) 0.4.1+3447790d18
		fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.1.6+cf39a74318
		fabric-dimensions-v1: Fabric Dimensions API (v1) 2.0.13+cf39a74318
		fabric-entity-events-v1: Fabric Entity Events (v1) 1.4.1+377137cc18
		fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.12+e99fbe1218
		fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.4+cf39a74318
		fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.8+cf39a74318
		fabric-item-api-v1: Fabric Item API (v1) 1.3.0+6617390918
		fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.2+cf39a74318
		fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.6+2a2bb57318
		fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.4+cf39a74318
		fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 1.4.6+0392f3a618
		fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.0.5+cf39a74318
		fabric-mining-level-api-v1: Fabric Mining Level API (v1) 1.0.3+cf39a74318
		fabric-mining-levels-v0: Fabric Mining Levels (v0) 0.1.6+cf39a74318
		fabric-models-v0: Fabric Models (v0) 0.3.1+cf39a74318
		fabric-networking-api-v1: Fabric Networking API (v1) 1.0.14+cf39a74318
		fabric-networking-blockentity-v0: Fabric Networking Block Entity (v0) 0.2.12+cf39a74318
		fabric-networking-v0: Fabric Networking (v0) 0.3.3+cf39a74318
		fabric-object-builder-api-v1: Fabric Object Builder API (v1) 1.11.0+f907116918
		fabric-object-builders-v0: Fabric Object Builders (v0) 0.7.7+cf39a74318
		fabric-particles-v1: Fabric Particles (v1) 0.2.5+cf39a74318
		fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.7.13+cf39a74318
		fabric-renderer-api-v1: Fabric Renderer API (v1) 0.4.5+cf39a74318
		fabric-renderer-indigo: Fabric Renderer - Indigo 0.4.9+cf39a74318
		fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.6+cf39a74318
		fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.1.6+cf39a74318
		fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 0.2.1+cf39a74318
		fabric-rendering-v0: Fabric Rendering (v0) 1.1.7+cf39a74318
		fabric-rendering-v1: Fabric Rendering (v1) 1.10.1+377137cc18
		fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.4.10+f09604ce18
		fabric-screen-api-v1: Fabric Screen API (v1) 1.0.5+cf39a74318
		fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.1.9+cf39a74318
		fabric-structure-api-v1: Fabric Structure API (v1) 1.1.15+cf39a74318
		fabric-tag-extensions-v0: Fabric Tag Extensions (v0) 1.2.3+cf39a74318
		fabric-textures-v0: Fabric Textures (v0) 1.0.7+cf39a74318
		fabric-tool-attribute-api-v1: Fabric Tool Attribute API (v1) 1.3.2+cf39a74318
		fabric-transfer-api-v1: Fabric Transfer API (v1) 1.5.3+cf39a74318
		fabricloader: Fabric Loader 0.12.12
		iris: Iris 1.1.3
		java: OpenJDK 64-Bit Server VM 16
		minecraft: Minecraft 1.17.1
		org_anarres_jcpp: jcpp 1.4.14
		org_joml_joml: joml 1.10.2
		org_slf4j_slf4j-api: slf4j-api 1.7.12
		sodium: Sodium 0.3.3+build.8
	Launched Version: fabric-loader-0.12.12-1.17.1
	Backend library: LWJGL version 3.2.2 build 10
	Backend API: Radeon RX 580 Series GL version 3.2.14761 Core Profile Forward-Compatible Context 21.30.23.01 30.0.13023.1012, ATI Technologies Inc.
	Window size: 1920x986
	GL Caps: Using framebuffer using OpenGL 3.2
	GL debug messages: 
	Using VBOs: Yes
	Is Modded: Definitely; Client brand changed to 'fabric'
	Type: Client (map_client.txt)
	Graphics mode: fancy
	Resource Packs: vanilla
	Current Language: English (US)
	CPU: 12x AMD Ryzen 5 3600 6-Core Processor

**Comment by warjort** [comment]
You are trying to use a 1.17.1 client

> Minecraft Version ID: 1.17.1

to connect to a 1.18 server?

> Server brand: BungeeCord (git:BungeeCord-Bootstrap:1.18-R0.1-SNAPSHOT:a9d75c5:1616) <- PaperSpigot

**Reply by xEchoes** [reply]
it happens on 1.17 servers as well

**Comment by warjort** [comment]
> java.lang.ArithmeticException: / by zero
at net.minecraft.class_761.method_22710(class_761.java:1153)

This method is WorldRenderer.render()

The only thing I can see doing a division by a variable which could be zero is a calculation of framerate 
where it will divide by "maxFps" from your options.txt

So  check that is a valid value.

**Reply by xEchoes** [reply]
ah, ill check that out, thanks

**Comment by warjort** [comment]
if its not that, try without iris or sodium (or any other rendering mod you have) to see if they are causing the problem.
